### DISCUSSION TITLE
Nuget Package Version Bumped for Unchanged Libraries

### DISCUSSION THREAD

**Discussion by czb182** [dotnet/aspnetcore#26014] [General] [answered]
Do we release new versions of libraries even if no changes?  It seems so.  For instance, I see version 3.1.8 of Microsoft.AspNetCore.Mvc.NewtonsoftJson.  I'm the curious type, so I come to see whats changed, so I check here
https://github.com/dotnet/aspnetcore/compare/v3.1.7...v3.1.8
I don't see any changes listed for that project, which I would have expected, so aside from potential build changes, it would be the same package I think.  That seems weird to me, why release a new version if nothing has changed, seems like you would just not publish a new version of that package for this core version release.

**Comment by davidfowl** [comment, marked as answer]
Yes we do.

cc @Pilchie

**Reply by Pilchie** [reply]
For 2.1 we only publish packages that are changed, but it's been very error prone, where we sometimes forget to actually include all the packages affected by a change.  Since there are many fewer packages in 3.1 with the shared framework, we decided to simplify and just publish all of them each time.

**Reply by czb182** [reply]
Thanks.  I understand reasoning.
